nixos/modules/services/networking/seafile.nix

```diff
@@ -162,137 +299,250 @@ in
+
+                echo Migrating ccnet database to MySQL
+                ${lib.getExe pkgs.sqlite} ${ccnetDir}/PeerMgr/usermgr.db ".dump" | \
+                  ${lib.getExe cfg.seahubPackage.python3} ${cfg.seahubPackage}/scripts/sqlite2mysql.py > ${ccnetDir}/ccnet.sql
+                ${lib.getExe pkgs.sqlite} ${ccnetDir}/GroupMgr/groupmgr.db ".dump" | \
+                  ${lib.getExe cfg.seahubPackage.python3} ${cfg.seahubPackage}/scripts/sqlite2mysql.py >> ${ccnetDir}/ccnet.sql
+                sed 's/ctime INTEGER/ctime BIGINT/g' -i ${ccnetDir}/ccnet.sql
+                sed 's/email TEXT, role TEXT/email VARCHAR(255), role TEXT/g' -i ${ccnetDir}/ccnet.sql
+                ${pkgs.mysql-client}/bin/mysql --database=ccnet_db < ${ccnetDir}/ccnet.sql
+
+                echo Migrating seafile database to MySQL
+                ${lib.getExe pkgs.sqlite} ${cfg.dataDir}/seafile.db ".dump" | \
+                  ${lib.getExe cfg.seahubPackage.python3} ${cfg.seahubPackage}/scripts/sqlite2mysql.py > ${cfg.dataDir}/seafile.sql
+                sed 's/owner_id TEXT/owner_id VARCHAR(255)/g' -i ${cfg.dataDir}/seafile.sql
+                sed 's/user_name TEXT/user_name VARCHAR(255)/g' -i ${cfg.dataDir}/seafile.sql
+                ${pkgs.mysql-client}/bin/mysql --database=seafile_db < ${cfg.dataDir}/seafile.sql
+
+                echo Migrating seahub database to MySQL
+                echo 'SET FOREIGN_KEY_CHECKS=0;' > ${seahubDir}/seahub.sql
+                ${lib.getExe pkgs.sqlite} ${seahubDir}/seahub.db ".dump" | \
+                  ${lib.getExe cfg.seahubPackage.python3} ${cfg.seahubPackage}/scripts/sqlite2mysql.py >> ${seahubDir}/seahub.sql
+                sed 's/`permission` , `reporter` text NOT NULL/`permission` longtext NOT NULL/g' -i ${seahubDir}/seahub.sql
+                sed 's/varchar(256) NOT NULL UNIQUE/varchar(255) NOT NULL UNIQUE/g' -i ${seahubDir}/seahub.sql
+                sed 's/,    UNIQUE (`user_email`, `contact_email`)//g' -i ${seahubDir}/seahub.sql
+                sed '/INSERT INTO `base_dirfileslastmodifiedinfo`/d' -i ${seahubDir}/seahub.sql
+                sed '/INSERT INTO `notifications_usernotification`/d' -i ${seahubDir}/seahub.sql
+                sed 's/DEFERRABLE INITIALLY DEFERRED//g' -i ${seahubDir}/seahub.sql
+                ${pkgs.mysql-client}/bin/mysql --database=seahub_db < ${seahubDir}/seahub.sql
+
+                echo "${cfg.seahubPackage.seafile-server.version}-mysql" > "${seafRoot}"/server-setup
+                echo Migration complete
             else
-                echo "Unsupported upgrade" >&2
+                echo "Unsupported upgrade: $installedMajor.$installedMinor to $pkgMajor.$pkgMinor" >&2
                 exit 1
             fi
           '';
+
+          postStart = ''
+            # Fix unix socket permissions
+            host="${cfg.seafileSettings.fileserver.host}"
+            if [[ "''${host:0:5}" =~ ^unix:.* ]]; then
```

Comment:
Can we do that condition in nix?

Reply:
I don't know why I didn't do it that way first, it's much cleaner now. I also made the nixos test use this codepath now.